lib/parse/util.rb

```diff
@@ -86,4 +88,26 @@ def Parse.object_pointer_hash(v)
       v.class_name.hash ^ v.id.hash
     end
   end
+
+
+  # NOTE: this mess is used to pass along the @client to internal objects
+  def Parse.copy_client(client, parsed_data)
```

Comment:
is decorating with this better than passing the client directly through parse_json?